internal/backend/mailbox.go

```diff
@@ -269,13 +303,17 @@ func (m *Mailbox) Expunge(ctx context.Context, seq *proto.SequenceSet) error {
 	var msg []*snapMsg
```

Comment:
```
	msg, err := snapshotReadErr(m.snap, func(s *snapshot) ([]*snapMsg, error) {
		if seq == nil {
			return s.getAllMessages(), nil
		}

		return s.getMessagesInRange(ctx, seq)
	})
	if err != nil {
		return err
	}

	return m.expunge(ctx, msg)
```

could also make this nicer with one call to `snapshotReadErr`

Reply:
Done